iothub/service/src/ServiceClientOptions.cs

```diff
@@ -0,0 +1,19 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using Microsoft.Azure.Devices.Shared;
+
+namespace Microsoft.Azure.Devices
+{
+    /// <summary>
+    /// Options that allow configuration of the service client instance during initialization.
+    /// </summary>
+    public class ServiceClientOptions
+    {
+        /// <summary>
+        /// The default configuration of MessageId for every message sent by the device or module client instance.
```

Comment:
I'm quibbling with the word default here. At first I thought we meant the default value on this options, but of course it can be changed so it is no longer default.

Then I realized you meant default value on `Message`, or at least I think that, so I think it should be clearly stated.

I'd suggest: `The default value of <see cref="Message.MessageId"/> for every message sent by the device or module client instance.`

Reply:
I see your point, but this isn't really the default value of the MessageId, but more of the default "configuration" - as it determines if the library will set a value for the messageId or not.

It is essentially this:
```csharp
public enum ShouldSdkAssignMessagaIdIfNotSet
{
    Yes,
    No
}
```

Reply:
This is what we want to get represented by the enum (nd matching description). Suggestions?

Reply:
My only issue here is on the doc comment

Reply:
Resolved:
```csharp
public enum SdkAssignsMessageId
{
    /// <summary>
    /// MessageId is set only by the user.
    /// </summary>
    Never,

    /// <summary>
    /// If MessageId is not set by the user, the client library will set it to a random GUID before sending the message.
    /// </summary>
    WhenUnset
}
```